DataFormats/L1THGCal/interface/HGCalClusterT.h

```diff
@@ -114,14 +115,18 @@ namespace l1t {
         auto id_fraction = constituentsFraction_.find(id_constituent.first);
         double fraction = (id_fraction != constituentsFraction_.end() ? id_fraction->second : 1.);
         if ((id.det() == DetId::Forward && id.subdetId() == HGCEE) || (id.det() == DetId::HGCalEE) ||
-            (id.det() == DetId::HGCalTrigger &&
-             HGCalTriggerDetId(id).subdet() == HGCalTriggerSubdetector::HGCalEETrigger)) {
+	    (id.det() == DetId::HGCalTrigger && HGCalTriggerDetId(id_constituent.first).subdet() == HGCalTriggerSubdetector::HGCalEETrigger) ||
+	    (id.det() == DetId::Forward && id.subdetId() == ForwardSubdetector::HFNose && HFNoseDetId(id_constituent.first).isEE()) ||
+	    (id.det() == DetId::HGCalTrigger && HGCalTriggerDetId(id_constituent.first).subdet() == HGCalTriggerSubdetector::HFNoseTrigger && HFNoseTriggerDetId(id_constituent.first).isEE())
+	    ) {
```

Comment:
- Mixed indents to be fixed
- `HGCalTriggerDetId(id)` (and similar) can be used instead of `HGCalTriggerDetId(id_constituent.first)`. More readable.

Reply:
this lead to the code crash

Reply:
I'm not sure to understand why it would crash. Is it in the `HFNoseTriggerDetId` constructor? There is an existing constructor from `DetId`, so if it crashes, there is something to be fixed elsewhere, but it is not a reason to change the code here.

Reply:
The original HGCalTriggerDetId( DetId(id_constituent.first) )  crash when a nose objects comes; HGCalTriggerDetId( id_constituent.first ) works because the rawId is passed directly.
I discussed with Sunanda once.

I apply in HGCalClusterT.h  what I did also in the in other parts, i.e. also in 
L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryV9Imp2.cc

Reply:
Do we know why it is crashing? I would prefer that it is fixed upstream.

Reply:
done, needs https://github.com/cms-sw/cmssw/pull/28249

Reply:
I backported the fix in #404